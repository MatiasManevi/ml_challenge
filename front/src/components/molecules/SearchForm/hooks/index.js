export { default as useSearchForm } from 'components/molecules/SearchForm/hooks/useSearchForm';
