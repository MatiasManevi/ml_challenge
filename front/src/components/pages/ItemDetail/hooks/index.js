export { default as useItemDetail } from 'components/pages/ItemDetail/hooks/useItemDetail';
