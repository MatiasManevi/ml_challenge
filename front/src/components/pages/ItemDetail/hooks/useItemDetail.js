import React from 'react';
import axios from 'axios';

import { API_BASE_URL } from 'utils';

const useItemDetail = (id) => {
	const [itemDetail, setItemDetail] = React.useState(null);

	React.useEffect(async () => {
		try {
			const response = await axios.get(`${API_BASE_URL}/items/${id}`);
			setItemDetail(response.data.item);
		} catch (e) {
			console.error(e);
		}
	}, []);

	return itemDetail;
};

export default useItemDetail;
