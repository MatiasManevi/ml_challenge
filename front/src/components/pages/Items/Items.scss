@import 'assets/scss/variables/spacing';

.items__link {
	text-decoration: none;
}
