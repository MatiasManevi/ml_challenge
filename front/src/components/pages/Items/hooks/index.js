export { default as useItems } from 'components/pages/Items/hooks/useItems';
