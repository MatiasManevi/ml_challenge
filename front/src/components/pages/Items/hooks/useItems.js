import React from 'react';

import { paginationQuery } from 'utils';
import ItemService from 'services/items';

const useItems = () => {
	const [items, setItems] = React.useState([]);
	const [categories, setCategories] = React.useState([]);
	const search = new URLSearchParams(window.location.search).get('search');

	React.useEffect(async () => {
		try {
			const response = await ItemService.get(search + paginationQuery);
			setItems(response.items);
			setCategories(response.categories);
		} catch (e) {
			console.error(e);
		}
	}, [search]);

	return {
		items,
		categories
	};
};

export default useItems;
