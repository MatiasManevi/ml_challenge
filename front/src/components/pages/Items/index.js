export { default } from 'components/pages/Items/Items';
