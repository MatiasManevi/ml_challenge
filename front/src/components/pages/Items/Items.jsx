import React from 'react';
import { Link } from 'react-router-dom';

import './Items.scss';
import { Item, BreadcrumbMenu } from 'components/organisms';
import { Content } from 'components/templates';
import { useItems } from './hooks';

const Items = () => {
	const { items } = useItems();

	return (
		<Content>
			<div className="items">
				<BreadcrumbMenu />
				{items.map((item) => (
					<Link key={item.id} className="items__link" to={`/items/${item.id}`}>
						<Item item={item} />
					</Link>
				))}
			</div>
		</Content>
	);
};

export default Items;
