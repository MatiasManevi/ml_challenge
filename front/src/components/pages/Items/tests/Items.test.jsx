import React from 'react';
import { shallow } from 'enzyme';

import Items from '../Items';
import * as hooks from '../hooks';
import items from './items.fixtures';

jest.mock('../hooks');

const stateSpy = jest.spyOn(hooks, 'useItems');

stateSpy.mockReturnValue({
	items
});

describe('Items', () => {
	it('should match snapshot', () => {
		const wrapper = shallow(<Items />);
		expect(wrapper).toMatchSnapshot();
	});
});
