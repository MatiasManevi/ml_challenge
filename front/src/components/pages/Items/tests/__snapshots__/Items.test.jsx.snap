// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items should match snapshot 1`] = `
<Content>
  <BreadcrumbMenu />
  <Link
    className="items__link"
    key="MLA923008999"
    to="/items/MLA923008999"
  >
    <Item
      item={
        Object {
          "condition": "new",
          "free_shipping": true,
          "id": "MLA923008999",
          "picture": "http://http2.mlstatic.com/D_712413-MLA46206646617_052021-O.jpg",
          "price": Object {
            "amount": 26399,
            "currency": "$",
            "decimals": "00",
          },
          "title": "Cacerola 24 Cm Essen Aqua Con Antiadherente ",
        }
      }
    />
  </Link>
  <Link
    className="items__link"
    key="MLA920404456"
    to="/items/MLA920404456"
  >
    <Item
      item={
        Object {
          "condition": "new",
          "free_shipping": true,
          "id": "MLA920404456",
          "picture": "http://http2.mlstatic.com/D_985487-MLA45983108078_052021-O.jpg",
          "price": Object {
            "amount": 75886,
            "currency": "$",
            "decimals": "00",
          },
          "title": "Essen Set Cacerola Sarten 24 Y Cacerola 18 Mas Utensilios",
        }
      }
    />
  </Link>
  <Link
    className="items__link"
    key="MLA839886481"
    to="/items/MLA839886481"
  >
    <Item
      item={
        Object {
          "condition": "new",
          "free_shipping": true,
          "id": "MLA839886481",
          "picture": "http://http2.mlstatic.com/D_750416-MLA44276051760_122020-O.jpg",
          "price": Object {
            "amount": 75950,
            "currency": "$",
            "decimals": "00",
          },
          "title": "Set Aqua Essen 3 Piezas + Set De Utensilios Super Oferta",
        }
      }
    />
  </Link>
  <Link
    className="items__link"
    key="MLA910856297"
    to="/items/MLA910856297"
  >
    <Item
      item={
        Object {
          "condition": "new",
          "free_shipping": true,
          "id": "MLA910856297",
          "picture": "http://http2.mlstatic.com/D_861354-MLA45116844792_032021-O.jpg",
          "price": Object {
            "amount": 62750,
            "currency": "$",
            "decimals": "00",
          },
          "title": "Essen Set Marsala 2 Piezas 28 Cm C/ant Nova Trimium Oferta!!",
        }
      }
    />
  </Link>
</Content>
`;
